salt/matchers/list_match.py

```diff
@@ -1,26 +1,27 @@
-# -*- coding: utf-8 -*-
 """
 This is the default list matcher.
 """
-from __future__ import absolute_import, print_function, unicode_literals
 
 import logging
 
 log = logging.getLogger(__name__)
 
 
-def match(tgt, opts=None):
+def match(tgt, opts=None, minion_id=None):
     """
     Determines if this host is on the list
     """
 
     if not opts:
```

Comment:
Can we remove this and update rest of `opts['id']` references to `minion_id` and the same goes for all the othe matchers. Not necessary to get merged.

Reply:
@Ch3LL so have `minion_id` in opts?

Reply:
Let me explain further.

```
if not opts:
    opts = __opts__
```

I'm not seeing why it is necessary because on the line below you query `id` from `__opts__` if it does not exist in `opts`, so I do not see the reason for keeping `opts = __opts__` when `minion_id` will correctly query the information as before. This is not necessary to merge, just something I observed, and I could be wrong here.

Reply:
Ahh yeah. that makes more sense now.  I updated the PR and pulled `opts` out as an argument to the various match functions.  There are still a few that need additional tests to ensure coverage.